app/assets/javascripts/admin.js

```diff
@@ -39,4 +38,7 @@ $(document).on('ready', function(e) {
     $feedback.find('.message').text(message);
     $feedback.show();
   });
+
+  $('select').not('.disabled').material_select();
+  Materialize.updateTextFields();
```

Comment:
I added comments to make this more clear. These functions reinitialize materialize  components which can display weirdly/not at all if the components are dynamically created. Im still not sure why this occurs, but it is advised in the Materialize documentation: http://materializecss.com/forms.html